mesonbuild/compilers/d.py

```diff
@@ -122,8 +122,16 @@ def get_feature_args(self, kwargs):
             import_dir_arg = d_feature_args[self.id]['import_dir']
             if not import_dir_arg:
                 raise EnvironmentException('D compiler %s does not support the "string import directories" feature.' % self.name_string())
-            for d in import_dirs:
-                res.append('{0}{1}'.format(import_dir_arg, d))
+            for idir_obj in import_dirs:
+                basedir = idir_obj.get_curdir()
+                for idir in idir_obj.get_incdirs():
+                     # Avoid superfluous '/.' at the end of paths when d is '.'
```

Comment:
__[Flake8]__

[E114] indentation is not a multiple of four (comment)


<!-- Comment by [SideCI](https://sideci.com) -->